### PR TITLE
Remove idle page

### PR DESCRIPTION
**Pull request by miherlosev**
Changes:
* set `[BrowserConnection.id]` title for each visited page. It's a temporary solution.  In future we will set the unique page title `<sessionId><windowId>` for each visited test page.
* in `BrowserConnectionGateway._onConnection` method redirects to the first tested page instead of the idle page (only for non-persistent browser connections).
* resize browser window on the first tested page during driver starting
* fix typos and remove unused code

Connected PRs:
* https://github.com/DevExpress/testcafe-hammerhead/pull/2406
* https://github.com/DevExpress/testcafe-legacy-api/pull/46

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Review by AlexKamaev — APPROVED** [review]

File: test/functional/fixtures/regression/gh-3049/test.js [hunk 1]
```diff
@@ -20,11 +19,11 @@ if (isLocalChrome) {
...
                         .run();
                 })
-                .then(err => {
+                .then(failedCount => {
                     testCafe.close();
 
-                    if (err)
-                        throw new Error('Error occured');
+                    if (failedCount)
+                        throw new Error('Error is occurred');
```
Comment (AlexKamaev) [inline test/functional/fixtures/regression/gh-3049/test.js:26]: The `Error is occurred` message is not correct.

Comment (miherlosev) [reply test/functional/fixtures/regression/gh-3049/test.js:26]: I will apply this change after Andrey review it.

File: package.json [hunk 2]
```diff
@@ -119,8 +119,8 @@
     "source-map-support": "^0.5.16",
     "strip-bom": "^2.0.0",
     "testcafe-browser-tools": "2.0.13",
-    "testcafe-hammerhead": "17.1.14",
-    "testcafe-legacy-api": "4.0.0",
+    "testcafe-hammerhead": "https://github.com/miherlosev/tmp/files/5154771/testcafe-hammerhead-17.1.14.zip",
```
Comment (AndreyBelym) [inline package.json:122]: Just a reminder to postpone merging until a npm versions are available.

**Review by AndreyBelym — APPROVED** [review]
An excellent job, congrats! But I will make it as a draft to avoid accidental merging until stable versions of dependencies are used.

**Pull request closed without merging**